### DISCUSSION TITLE
Inconsistent behavior from GlueJobOperator

### DISCUSSION THREAD

**Discussion by caldowa** [apache/airflow#30234] [General]
I'm currently trying to migrate the amazon provider from version `6.0.0` to version `7.3.0` to benefit from bug fixes and new features.

While testing the new version of the provider before deploying to prod, I've noticed an inconsistent behavior from the GlueJobOperator.

The GlueJobOperator is used to run a Glue Job on AWS and creates the job if it doesn't already exists. You can pass in parameters for the job creation via the argument `create_job_kwargs` like in the following code snippet : 

```python
GlueJobOperator(
    task_id="<my_task_id>",
    job_name="<glue_job_name>",
    wait_for_completion=True,
    script_location="s3://<script_location>",
    s3_bucket=glue_bucket,
    iam_role_name=glue_iam_role,
    create_job_kwargs={
        "GlueVersion": "3.0",
        "WorkerType": "G.1X",
        "NumberOfWorkers": 4,
        "DefaultArguments": {
            "--execution-class": "FLEX",
            "--enable-auto-scaling": "true",
            "--enable-glue-datacatalog": "true",
        },
        "Tags": {
            "scope": "transformation",
            "env": env,
            "created_by": "Astronomer",
        },
    },
)
```

If the job doesn't exist it will create it with said arguments and run it with no problem. However, if the job already exists an error gets thrown because the aws-sdk doesn't accept updating Glue Tags.

Here is the error message : 

`botocore.exceptions.ParamValidationError: Parameter validation failed:
Unknown parameter in JobUpdate: "Tags", must be one of: Description, LogUri, Role, ExecutionProperty, Command, DefaultArguments, NonOverridableArguments, Connections, MaxRetries, AllocatedCapacity, Timeout, MaxCapacity, WorkerType, NumberOfWorkers, SecurityConfiguration, NotificationProperty, GlueVersion, CodeGenConfigurationNodes, ExecutionClass`

It seems this was [done by design](https://github.com/aws/aws-sdk/issues/91) so I'm not really sure whether to open an issue regarding the provider or not ?

It seems a bit inconsistent to be able to pass in Tags when the job isn't created, for the operator to then throw an error if used with an existing job?

**Comment by kobethuwis** [comment]
Same 'design' is actually present in the GlueCrawlerOperator (without the error though). Tags are updated when a crawler exists for a specified name; which is weird because you shouldn't have to specify tags to retrieve a crawler by name.

Actually had to write a fix for that since it was updating our terraform-deployed crawlers everytime a crawler was ran, so now we're dynamically setting these tags in the config of the crawler:
```
def get_crawler_operator(name, dag, suffix=0):
    from helpers.environment_helper import get_refresh_environment
    config = {"Name": name, "Tags": {"Terraform": f'datalake-{get_refresh_environment().lower()}'}}
    task_id = name
    if suffix != 0:
        task_id += f'-{str(suffix)}'
    from operators.crawler_operator import CustomCrawlerOperator
    glue_crawler_task = CustomCrawlerOperator(
        config=config, task_id=task_id, dag=dag
    )
    glue_crawler_task.region_name = 'eu-west-1'
    return glue_crawler_task
```
